presto-spi/src/main/java/com/facebook/presto/spi/block/AbstractMapElementBlock.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.spi.block;
+
+import io.airlift.slice.Slice;
+
+import java.util.List;
+
+public abstract class AbstractMapElementBlock
+        implements Block
+{
+    private final int offset;
+    private final Block keyBlock;
+    private final Block valueBlock;
+
+    public AbstractMapElementBlock(int offset, Block keyBlock, Block valueBlock)
+    {
+        this.offset = offset;
+        this.keyBlock = keyBlock;
+        this.valueBlock = valueBlock;
+    }
+
+    private int getAbsolutePosition(int position)
+    {
+        if (position < 0 || position >= getPositionCount()) {
+            throw new IllegalArgumentException("position is not valid");
+        }
+        return position + offset;
+    }
+
+    @Override
+    public boolean isNull(int position)
+    {
+        position = getAbsolutePosition(position);
+        if (position % 2 == 0) {
+            return keyBlock.isNull(position / 2);
```

Comment:
It is changed to throw an exception when `keyBlock.isNull(...)` evaluates to true.